main/src/ca/uwaterloo/flix/language/ast/ErasedAst.scala

```diff
@@ -357,5 +357,67 @@ object ErasedAst {
     sealed trait JChan[T <: JType] extends JType
     sealed trait JLazy[T <: JType] extends JType
   }
+
+  sealed trait ErasedType[+T <: JType]
+
+  object ErasedType {
+
+    case class Unit() extends ErasedType[JUnit]
+
+    case class Bool() extends ErasedType[PrimInt32]
+
+    case class Char() extends ErasedType[PrimChar]
+
+    case class Float32() extends ErasedType[PrimFloat32]
+
+    case class Float64() extends ErasedType[PrimFloat64]
+
+    case class Int8() extends ErasedType[PrimInt8]
+
+    case class Int16() extends ErasedType[PrimInt16]
+
+    case class Int32() extends ErasedType[PrimInt32]
+
+    case class Int64() extends ErasedType[PrimInt64]
+
+    case class BigInt() extends ErasedType[JObject]
+
+    case class Str() extends ErasedType[JObject]
+
+    ///
+    /// Compound Types.
+    ///
+
+    case class Array[T <: JType](tpe: ErasedType[T]) extends ErasedType[JArray[T]]
+
+    case class Channel[T <: JType](tpe: ErasedType[T]) extends ErasedType[JChan[T]]
+
+    case class Lazy[T <: JType](tpe: ErasedType[T]) extends ErasedType[JLazy[T]]
+
+    case class Ref[T <: JType](tpe: ErasedType[T]) extends ErasedType[JRef[T]]
+
+    case class Tuple(elms: List[ErasedType[JType]]) extends ErasedType[JObject]
+
+    case class Enum(sym: Symbol.EnumSym, args: List[ErasedType[JType]]) extends ErasedType[JObject]
+
+    case class Arrow(args: List[ErasedType[JType]], result: ErasedType[JType]) extends ErasedType[JObject]
+
+    case class RecordEmpty() extends ErasedType[JObject]
+
+    case class RecordExtend(field: String, value: ErasedType[JType], rest: ErasedType[JObject]) extends ErasedType[JObject]
+
+    case class SchemaEmpty() extends ErasedType[JObject]
+
+    case class SchemaExtend(name: String, tpe: ErasedType[JType], rest: ErasedType[JObject]) extends ErasedType[JObject]
+
+    case class Relation(tpes: List[ErasedType[JType]]) extends ErasedType[JObject]
+
+    case class Lattice(tpes: List[ErasedType[JType]]) extends ErasedType[JObject]
+
+    case class Native(clazz: Class[_]) extends ErasedType[JObject]
+
+    // TODO: Should be removed.
+    case class Var(id: Int) extends ErasedType[Nothing]
```

Comment:
Where is this used? I think we should throw an exception if the Eraser encounters a type var.

Reply:
OK, but I guess if this crashes the compiler then I need to fix it elsewhere first.

Reply:
There is still vars being given when running tests, so im keeping it in ErasedType